consensus/leader_election/module.go

```diff
@@ -60,5 +60,14 @@ func (m *leaderElectionModule) ElectNextLeader(message *typesCons.HotstuffMessag
 
 func (m *leaderElectionModule) electNextLeaderDeterministicRoundRobin(message *typesCons.HotstuffMessage) typesCons.NodeId {
 	value := int64(message.Height) + int64(message.Round) + int64(message.Step) - 1
-	return typesCons.NodeId(value%int64(len(m.GetBus().GetConsensusModule().ValidatorMap())) + 1)
+
+	ctx, err := m.GetBus().GetPersistenceModule().NewReadContext(int64(message.Height))
+	if err != nil {
+		log.Fatal(err)
+	}
+	vals, err := ctx.GetAllValidators(int64(message.Height))
```

Comment:
Shouldn't this be part of `issue/270-p2p-integration-with-utility-for-addrbook-lookups`? I'm a little confused with the two different branches are dependant on each other.

Reply:
You are right, reverted.

The branches are dependent because from my understanding there were overlaps.

My confusion stems also from this https://github.com/pokt-network/pocket/pull/374#discussion_r1042044513 

What I mean is: If we have to access `GetAllValidators` from the context within `Utility`, shouldn't we simply integrate `P2P` with `Utility` (#270)?

Lemme know if this clarifies your confusion about my confusion :)